vignettes/layer_templates.Rmd

```diff
@@ -0,0 +1,121 @@
+
+Tplyr has already has a couple mechanisms to reduce redundant application of formats. For example, vignettes('tplyr_options') shows how the options `tplyr.count_layer_default_formats`, `tplyr.desc_layer_default_formats`, and `tplyr.shift_layer_default_formats` can be used to create default format string settings. Additionally, you can set formats table wide using `set_count_layer_formats()`, `set_desc_layer_formats()`, or `set_shift_layer_formats()`. But what these functions and options _don't_ allow you to do is pre-set and reuse the settings for an entire layer, so all of the additional potential layer modifying functions are ignored. This is where layer templates come in.
+
+# Basic Templates 
+
+The functions `new_layer_template()` and `use_template()` allow a user to create and use layer templates. Layer templates allow a user to pre-build and reuse an entire layer configuration, from the layer constructor down to all modifying functions. Furthermore, users can specify parameters they may want to be interchangeable. Additionally, layer templates are extensible, so a template can be use and then further extended with additional layer modifying functions.
+
+Consider the following example:
+
+```{r creating a template}
+new_layer_template(
+  "example_template", 
+  group_count(...) %>% 
+    set_format_strings(f_str("xx (xx%)", n, pct))
+)
+```
+
+In this example, we've created a basic layer template. The template is named "example_template", and this is the name we'll use to reference the template when we want to use it. When the template is created, we start with the function `group_count(...)`. Note the use of the ellipsis (i.e. `...`). This is a required part of a layer template. Templates must start with a Tplyr layer constructor, which is one of the function `group_count()`, `group_desc()`, or `group_shift()`. The ellipsis is necessary because when the template is used, we are able to pass arguments directly into the layer constructor. For example:
+
+```{r using a template}
+tplyr_table(adsl, TRT01P) %>% 
+  add_layer(
+    use_template("example_template", RACE, by=ETHNIC)
+  ) %>% 
+  build() %>% 
+  kable()
+```
+
+Within `use_template()`, the first parameter is the template name. After that, we supply arguments as we normally would into `group_count()`, `group_desc()`, or `group_shift()`. Additionally, note that our formats have been applied just as they would be if we used `set_format_strings()` as specified in the template. Our template was a applied, the table built with all of the settings appropriately. 
```

Comment:
Extra 'a' in 'Our template was *a* applied'.

Reply:
Updated